server/src/main/java/org/opensearch/common/bytes/AbstractBytesReference.java

```diff
@@ -82,7 +84,13 @@ public void writeTo(OutputStream os) throws IOException {
 
     @Override
     public String utf8ToString() {
-        return toBytesRef().utf8ToString();
+        BytesRef bytesRef = toBytesRef();
+        final char[] ref = new char[bytesRef.length];
+        final int len = UnicodeUtil.UTF8toUTF16(bytesRef.bytes, bytesRef.offset, bytesRef.length, ref);
```

Comment:
The only concern I have is, this might generate garbage as full characters will have to be read. So we will need allocation profiling samples. Is there a way to handle OOM resulting from the current problem at hand, more gracefully.

Reply:
The step was already done in BytesRef - 
```
public String utf8ToString() {
    final char[] ref = new char[length];
    final int len = UnicodeUtil.UTF8toUTF16(bytes, offset, length, ref);
    return new String(ref, 0, len);
  }
```

This change adds a pre-check on length after the moving this conversion to `AbstractBytesReference.java`

Reply:
I was having the same concern as @Bukhtawar but it seems like there are no available means (at least within UnicodeUtil) to estimate how much the converted string might take ...

Reply:
So this seems like there's actually no new garbage being created. @Bukhtawar are you saying otherwise?